hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -8408,42 +8423,21 @@ public void startRegionOperation() throws IOException {
 
   @Override
   public void startRegionOperation(Operation op) throws IOException {
-    boolean isInterruptableOp = false;
-    switch (op) {
-      case GET: // interruptible read operations
-      case SCAN:
-        isInterruptableOp = true;
-        checkReadsEnabled();
-        break;
-      case INCREMENT: // interruptible write operations
-      case APPEND:
-      case PUT:
-      case DELETE:
-      case BATCH_MUTATE:
-      case CHECK_AND_MUTATE:
-        isInterruptableOp = true;
-        break;
-      default: // all others
-        break;
-    }
-    if (
-      op == Operation.MERGE_REGION || op == Operation.SPLIT_REGION || op == Operation.COMPACT_REGION
-        || op == Operation.COMPACT_SWITCH
-    ) {
-      // split, merge or compact region doesn't need to check the closing/closed state or lock the
-      // region
-      return;
-    }
+    boolean isInterruptibleOp = isOperationInterruptible(op);
+    if (!shouldLock(op)) return;
```

Comment:
I feel this has improved readibility but i am fine if someone wants to revert this.

Reply:
Yes, this too was simpler in the earlier version of the PR.